case_test.go

```diff
@@ -0,0 +1,110 @@
+package squirrel
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestCaseWithVal(t *testing.T) {
+	caseStmt := Case().
+		What("number").
```

Comment:
I've tried to avoid introducing method names that don't correspond to SQL keywords . Could it be done in the Case() method instead? Basically:

```go
func Case(args ...interface{}) CaseBuilder {
    ...
    if len(args) > 0 {
       b = b.what(newPart(args...))
    }
    ... 
}